### PR TITLE
Add recipe for power-ci

### PR DESCRIPTION
**Pull request by leel8on**
Initial work for power-ci.o.o. Installs Jenkins, Jenkins plugins, and haproxy. 

The list of Jenkins plugins is taken from https://power-ci.osuosl.org/jenkins/

Because of a delay when Jenkins is restarting, kitchen test will most likely fail. Running kitchen verify again after the failure will usually work.

### CODE REVIEW

**Comment by leel8on** [comment]
[powerci recipe in osl-jenkins](https://trello.com/c/v7DGMSQS/441-powerci-recipe-in-osl-jenkins)

**Comment by leel8on** [comment]
There is also a merge request for the power-ci cert data bag:

https://git.osuosl.org/osuosl-chef/data_bags/merge_requests/287

**Comment by leel8on** [comment]
@ramereth Did you want me to use the wildcard certificate attributes, haproxy config, and the git/ssh credentials from the master recipe? Or should I change those as well? 

For the certs, I think we got the power-ci certs during senior project since this was all being done outside the normal chef managed infra. But I'll let you decide if we want to just use the wildcard cert now.

**Comment by ramereth** [comment]
@leel8on let's use the wildcard for now since its easier.

**Comment by leel8on** [comment]
@ramereth What about the following from the haproxy recipe?
```
node.default['haproxy']['members'] = [
  {
    'hostname' => 'jenkins',
    'ipaddress' => '127.0.0.1',
    'port' => '8080'
  }
]
```
Should that become
```
node.default['haproxy']['members'] = [
  {
    'hostname' => 'power-ci',
    'ipaddress' => '127.0.0.1',
    'port' => '8080'
  }
]
```
I don't think I've setup haproxy before.

**Comment by leel8on** [comment]
@ramereth updated to use the master recipe.

File: recipes/powerci.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# limitations under the License.
+#
+
+node.override['haproxy']['members'] = [
+  {
+    'hostname' => 'power-ci',
+    'ipaddress' => '127.0.0.1',
+    'port' => '8080'
+  }
+]
```
Comment (ramereth) [inline recipes/powerci.rb]: This isn't needed. The default should just work AFAIK.

File: recipes/powerci.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+node.default['osl-jenkins']['restart_plugins'] = %w(
+  credentials:2.1.13
+  ssh-credentials:1.13
+)
```
Comment (ramereth) [inline recipes/powerci.rb]: This isn't needed since it's already covered in the default attributes.

**Comment by leel8on** [comment]
Updated to remove unneeded sections.

File: spec/unit/recipes/powerci_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(p).converge(described_recipe)
+      end
+      include_context 'common_stubs'
+      it 'converges successfully' do
+        expect { chef_run }.to_not raise_error
+      end
+      it do
+        expect(chef_run).to install_package('jenkins').with(version: '2.46.3-1.1')
+      end
```
Comment (ramereth) [inline spec/unit/recipes/powerci_spec.rb]: This is not needed as its covered in the master spec.

File: spec/unit/recipes/powerci_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      case p
+      when CENTOS_6_OPTS
+        it do
+          expect(chef_run).to create_link('/usr/bin/git').with(to: '/usr/local/bin/git')
+        end
+      end
```
Comment (ramereth) [inline spec/unit/recipes/powerci_spec.rb]: This is not needed as its covered in the master spec.

File: test/integration/powerci/serverspec/powerci_spec.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
+require 'spec_helper'
+
+set :backend, :exec
+
+describe service('haproxy') do
+  it { should be_running }
+end
```
Comment (ramereth) [inline test/integration/powerci/serverspec/powerci_spec.rb]: This is not needed.

File: .kitchen.yml [hunk 6]
```diff
@@ -246,3 +246,7 @@ suites:
       jenkins:
         master:
           listen_address: '0.0.0.0'
+  - name: powerci 
+    encrypted_data_bag_secret_key_path: test/integration/encrypted_data_bag_secret
```
Comment (ramereth) [inline .kitchen.yml]: This actually isn't needed.

File: spec/unit/recipes/powerci_spec.rb (hunk 4 above)
Comment (leel8on) [inline spec/unit/recipes/powerci_spec.rb]: @ramereth Is there anything other than converging successfully that you think should be checked in the chefspec?

Comment (ramereth) [inline spec/unit/recipes/powerci_spec.rb]: You might check _some_ (but not all) of the plugins that are critical to this environment are being installed and with the correct version.

**Comment by ramereth** [comment]
@leel8on actually, I somewhat lied. I would check any plugins that aren't already checked when you run the full rspec suite. I.e. I still see the following needing tests:

```
Untouched Resources:

  jenkins_plugin[resource-disposer]   osl-jenkins/recipes/plugins.rb:32
  jenkins_plugin[emailext-template]   osl-jenkins/recipes/plugins.rb:32
  jenkins_plugin[openstack-cloud]    osl-jenkins/recipes/plugins.rb:32
  jenkins_plugin[config-file-provider]   osl-jenkins/recipes/plugins.rb:32
  jenkins_plugin[email-ext]          osl-jenkins/recipes/plugins.rb:32
  jenkins_plugin[pipeline-multibranch-defaults]   osl-jenkins/recipes/plugins.rb:32
  jenkins_plugin[cloud-stats]        osl-jenkins/recipes/plugins.rb:32
```

**Comment by leel8on** [comment]
@ramereth Yeah those are missing from the plugins spec. Should I add those there or in the powerci spec?

**Comment by leel8on** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v1.6.0.  Have a nice day!

**Pull request merged**